crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -1455,13 +1457,39 @@ impl<T: PoolTransaction> AllTransactions<T> {
     pub(crate) fn remove_transaction_by_hash(
         &mut self,
         tx_hash: &B256,
-    ) -> Option<(Arc<ValidPoolTransaction<T>>, SubPool)> {
+        update_descendant: bool,
+    ) -> Option<(Arc<ValidPoolTransaction<T>>, SubPool, Vec<PoolUpdate>)> {
```

Comment:
instead of making this conditional I'd prefer having a new dedicated function for this instead, that we can call after `remove_transaction_by_hash` in `fn remove_transaction(`

Reply:
👌